### ISSUE TITLE
TFLite android example custom model error: java.lang.ArrayIndexOutOfBoundsException: length=10; index=-2147483646

### ISSUE THREAD

**Issue by shauryad15** [tensorflow/tensorflow#34374]
Hi @achowdhery ,

I'm getting this error while placing my custom model in tensorflow lite android example. I've gone through the post which resolves the same 
int labelOffset = 1;
      recognitions.add(
          new Recognition(
              "" + i,
              labels.get((int) outputClasses[0][i] + labelOffset),
              outputScores[0][i],
              detection));

this is the part which is throwing the error. But i'm not getting how to resolve it.

I annotated images using labelimg tool with class names as parallel_parking and perpendicular parking. Then while creating tf.record I did this:

item {
  id: 1
  name: 'parallel_parking'
}
item {
  id: 2
  name: 'perpendicular_parking'
}

and followed the steps mentioned here https://medium.com/tensorflow/training-and-serving-a-realtime-mobile-object-detector-in-30-minutes-with-cloud-tpus-b78971cf1193.

now, while running the tflite app, i'm getting the above mentioned error. Need help.

**Comment by jvishnuvardhan** [comment]
@shauryad15 Sorry for late response. Is it still an issue? Can you check whether this [resource](https://github.com/tensorflow/tensorflow/issues/22106#issuecomment-428409506) will help you in resolving your issue? If it is not resolving, please share a standalone code to reproduce your issue. Thanks!

**Comment by jvishnuvardhan** [comment]
@shauryad15 Is it still an issue? Please check the resource in my last response. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Ehtasha** [comment]
@jvishnuvardhan   I'm also facing this problem while running custom model. 

I'm integrating Custom Nail Detection Segmentation model with android app.

[Model tflite file link:](https://github.com/makeml-app/MakeML-Nails/tree/master/Segmentation%20Nails/Resources)

[Android Code which I'm using](https://github.com/sercant/android-segmentation-app) 

Above android code link works well with the model which is give but whenever I use MakeML nail detection model app crash with the error message.

`java.lang.ArrayIndexOutOfBoundsException: length=66049; index=66049` 

Error in bellow while loop.
`while (segmentedImage.hasRemaining()){
                outFrame[i++] = segmentedImage.int
        }` 


I have print the value of outframe use `outframe.size` and this value is 66049.
 
Already spend full week on this issue but no luck. Please guide me how to resolve this issue ?

**Comment by jvishnuvardhan** [comment]
@shauryad15 Can you please create a new issue with the details you mentioned above and any standalone code to reproduce the issue? Thanks!

**Comment by Ehtasha** [comment]
I have created separate issue on [this](https://github.com/tensorflow/tensorflow/issues/36956) link.

I have upload the stand alone code on [this](https://drive.google.com/open?id=1WQ9LmAqJpxMwZIHtzMxRP2j6nq1Ga2lD) link which produce this error.

This code is working fine with other models which come with the code.

**Comment by shauryad15** [comment]
Hi @Ehtasha/ @jvishnuvardhan  ,

This issue generally occurs when you're trying to read the index which isn't not present. TO solve this, please check your labels file and see whether you're accessing the correct index of the label or not. If you try to access the memory which isn't allocated, you'll get the error.

For ex: label 0 is Cat, label 1 is Dog. If access label 2 which isn't there we'll get the same error.

**Comment by Ehtasha** [comment]
@shauryad15  Thanks for reply.
You are right this is because of the index which don't exist. But in this case I don't have label file.

This model is doing segmentation on nails(Placing colors on nails ). There is only single tflite file which I'm using. 

This model is working fine with the IoS app and label file is also not present there. You can see [this](https://github.com/makeml-app/MakeML-Nails/tree/master/Segmentation%20Nails/Resources) link.

Other segmentation models which are working fine in android app also don't have label file.

**Comment by Muhammad-Ullah** [comment]
But can you please share that how to get the labels.txt file in while converting the keras model to tflite?
